Documentation updates for DataPrepper 2.8 (#7135)
opensearch-project/documentation-website: _data-prepper/pipelines/expression-syntax.md

null != /response
```

## Type check operator

> Type check operator tests if a JSON Pointer is of a certain type

Reviewer: ```suggestion

The type check operator tests whether a JSON Pointer is of a certain data type.
```

Reviewer: Should this read "The `typeof` operator..."?

### Syntax
```
<JSON Pointer> typeof <DataType>
```

> Supported <DataType> are `integer`, `long`, `boolean`, `double`, `string`, `map`, and `array`.

Reviewer: ```suggestion
Supported data types are `integer`, `long`, `boolean`, `double`, `string`, `map`, and `array`.
```

> ### Example

Reviewer: ```suggestion
#### Example
```

```
/response typeof integer
/message typeof string
```

> #### Conditional expression

Reviewer: ```suggestion
### Conditional expression
```

| `==`, `!=`           | Equality operators       | No                   | `/status == 200`<br>`/status_code==200`                        |                                       |
| `and`, `or`, `not`   | Conditional operators    | Yes                  | `/a<300 and /b>200`                                            | `/b<300and/b>200`                     |
| `,`                  | Set value delimiter      | No                   | `/a in {200, 202}`<br>`/a in {200,202}`<br>`/a in {200 , 202}` | `/a in {200,}`                        |

> | `typeof`             | type check operator      | Yes                   | `/a typeof integer`<br>`/a typeof long`<br>`/a typeof string`<br> `/a typeof double`<br> `/a typeof boolean`<br>`/a typeof map`<br>`/a typeof array` |`/a typeof /b`<br>`/a typeof 2`                      |

Reviewer: ```suggestion
| `typeof`             | Type check operator      | Yes                   | `/a typeof integer`<br>`/a typeof long`<br>`/a typeof string`<br> `/a typeof double`<br> `/a typeof boolean`<br>`/a typeof map`<br>`/a typeof array` |`/a typeof /b`<br>`/a typeof 2`                      |
```
